tests/queries/0_stateless/02302_column_decl_null_before_defaul_value.reference

```diff
@@ -0,0 +1,20 @@
+create table, column +type +NULL
+id	Nullable(Int32)					
+create table, column +type +NOT NULL
+id	Int32					
+create table, column +type +NULL +DEFAULT
+id	Nullable(Int32)	DEFAULT	1			
+create table, column +type +NOT NULL +DEFAULT
+id	Int32	DEFAULT	1			
+create table, column +type +DEFAULT +NULL
+id	Nullable(Int32)	DEFAULT	1			
+create table, column +type +DEFAULT +NOT NULL
+id	Int32	DEFAULT	1			
+create table, column -type +NULL +DEFAULT
+id	UInt8	DEFAULT	1			
```

Comment:
Here, type should be nullable, i.e. `Nullable(Int8)` instead of just `Int8`

I guess the issue is that in interpreter, we check if a column is nullable _only_ if type is present for the column. But here, only the default expression is provided

See https://github.com/ClickHouse/ClickHouse/blob/dea1706d4e65572a23636483bb8800f3c733a460/src/Interpreters/InterpreterCreateQuery.cpp#L478